### PR TITLE
[Tests] Fix intro tests on older macOS

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Comment by mandel-macaque** [comment]
I have tested all of this on:

* Mojave
* Catalina
* High Sierra
* Big Sur

**Review by dalexsoto — APPROVED** [review]

File: tests/introspection/Mac/MacApiSelectorTest.cs [hunk 1]
```diff
@@ -138,6 +138,13 @@ protected override bool Skip (Type type, string selectorName)
 					if (!Mac.CheckSystemVersion (10, 12)) // NSCoding was added in 10.12
 						return true;
 					break;
+				case "MLDictionaryFeatureProvider":
+				case "MLMultiArray":
+				case "MLFeatureValue":
+				case "MLSequence":
+					if (Mac.CheckSystemVersion (10, 15)) // Fail on Catalina, pass in older OS
+						return true;
```
Comment (spouliot) [inline tests/introspection/Mac/MacApiSelectorTest.cs:146]: that's skipping OS <= 10.15
if it pass on older you need a different check, like
```csharp
// Fail on Catalina, pass in older OS	
if (Mac.CheckSystemVersion (10, 14))
	break;
if (Mac.CheckSystemVersion (10, 15)) 
	return true;
// 11.x
break;
```

File: tests/introspection/Mac/MacApiProtocolTest.cs [hunk 2]
```diff
@@ -34,7 +34,14 @@ protected override bool Skip (Type type)
...
 			case "AVCaptureSynchronizedDataCollection":
 			case "AVCaptureSynchronizedData":
+			case "MPSImageLaplacianPyramid":
+			case "MPSImageLaplacianPyramidSubtract":
+			case "MPSImageLaplacianPyramidAdd":
+			case "MPSCnnYoloLossNode":
+			case "CXProvider":
 				return TestRuntime.IsVM; // skip only on vms
+			case "NSMenuView":
+				return TestRuntime.IsVM || IntPtr.Size != 8; // skip on vms or 64b
```
Comment (spouliot) [inline tests/introspection/Mac/MacApiProtocolTest.cs:44]: we don't support 32bits for macOS anymore, you can always skip `NSMenuView`

File: tests/introspection/Mac/MacApiCtorInitTest.cs [hunk 3]
```diff
@@ -200,7 +200,10 @@ protected override bool Skip (Type type)
 				return TestRuntime.CheckXcodeVersion (11, 2);
 			case "GameKit.GKGameCenterViewController": // the native 'init' method returned nil.
 				return TestRuntime.CheckXcodeVersion (11, 2);
-
+			case "MetalPerformanceShaders.MPSPredicate":
+				if (Mac.CheckSystemVersion (10, 15)) // Fails on Catalina: Could not initialize an instance of the type 'MetalPerformanceShaders.MPSPredicate': the native 'init' method returned nil. 
+					return true;
```
Comment (spouliot) [inline tests/introspection/Mac/MacApiCtorInitTest.cs:205]: does it work on earlier macOS ? 
if so see my other comment

**Review by spouliot — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

**Pull request merged**